// voicevox-api-client/dist/Client/Other.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OtherClient = void 0;
class OtherClient {
    constructor(client) {
        this.client = client;
    }
    margeWav(base64Array) {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield this.client.request('/connect_waves', {
                body: JSON.stringify(base64Array),
            });
            return res.arrayBuffer();
        });
    }
    getPreset() {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield this.client.request('/preset', {}, 'GET');
            return yield res.json();
        });
    }
    getVersion() {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield this.client.request('/version', {}, 'GET');
            return yield res.text();
        });
    }
    getSpeakers() {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield this.client.request('/speakers', {}, 'GET');
            return yield res.json();
        });
    }
    getSpeakerInfo(speaker_uuid) {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield this.client.request(`/speaker_info?speaker_uuid=${speaker_uuid}`, {}, 'GET');
            return yield res.json();
        });
    }
}
exports.OtherClient = OtherClient;
